### DISCUSSION TITLE
netdata-vs-prometheus-performance-analysis/

### DISCUSSION THREAD

**Discussion by giscus** [netdata/blog#340] [General]
# netdata-vs-prometheus-performance-analysis/

Explore the performance of Netdata when it comes to monitoring a huge Infrastructure.

https://blog.netdata.cloud/netdata-vs-prometheus-performance-analysis/

**Comment by fenio** [comment]
Nice results. I'd love to see comparison vs VictoriaMetrics too.

**Comment by hagen1778** [comment]
> The Netdata Parent was receiving all metrics in real-time from the 500 nodes, via Netdata streaming (they were pushing metrics to Netdata Parent).

Does it mean that metrics collectio&parsing was already done by some external agent and then pushed to Netdata parent? While Prometheus did the collection&parsing&storing - all done by the same process.

If it is so, then you need to include resource used by the external clients doing the streaming.

The same goes for network bandwidth - prometheus scrapes metrics in a plain text, while streaming metrics can use any custom protocol and compression. It would be more fair to compare Prometheus RemoteWrite protocol vs Netdata streaming.

> Netdata was configured to have 3 TB of space, which gives us a variable retention depending on how much data can fit in this storage space. 

This is confusing... Especially for people not familiar with storage tiering. Why not using one tier for transparency?

> Nice results. I'd love to see comparison vs VictoriaMetrics too.

Agree! VictoriaMetrics supports ingestion out-of-the-box, so it would be more fair to compare with Netdata streaming. It would be even better to use the same protocol for metrics delivery for both databases. Let me know if I can help in conducting such a benchmark.

**Comment by BenoitPoulet** [comment]
This study is biased, of course it's came from people who wrote Netdata, published on the Netdata website.

In my opinion, the right test must be done with Prometheus remote write, because in the case of Netdata, the Parent do nothing but storing the metrics.
